src/modules/yield-rates/yield-rates.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable, InternalServerErrorException, Logger, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { DATABASE } from '@ukef/constants';
+import { Equal, LessThanOrEqual, MoreThan, Repository } from 'typeorm';
+
+import { DATE } from '../../constants/date.constant';
+import { YieldRateEntity } from './entities/yield-rate.entity';
+
+@Injectable()
+export class YieldRatesService {
+  private readonly logger = new Logger();
+  constructor(
+    @InjectRepository(YieldRateEntity, DATABASE.CEDAR)
+    private readonly yieldRateRepository: Repository<YieldRateEntity>,
+  ) {}
+
+  async find(searchDate: string): Promise<YieldRateEntity[]> {
+    try {
+      // Search by date or use standard max date to get active/current rates.
+      const query: object = searchDate
+        ? { effectiveTo: MoreThan(searchDate), effectiveFrom: LessThanOrEqual(searchDate) }
+        : { effectiveTo: Equal(DATE.MAXIMUM_LIMIT) };
+
+      const results = await this.yieldRateRepository.find({
+        where: query,
+      });
+      if (results && !results[0]) {
```

Comment:
results will always be an array (either empty or with values). I think we can use here:
```js
if (!results.length) {...}
```

Reply:
Done. Tested and changed for all TypeORM responses. I left Stored Procedure response checks unchanged, maybe SP can return error object, not array.